src/Security/Certificate.cs

```diff
@@ -263,19 +270,47 @@ public NSData GetPublicKey ()
 			}
 		}
 #else
-		[iOS (10,3)]
-		[TV (10,3)]
-		[Deprecated (PlatformName.iOS, 12,0)]
-		[Deprecated (PlatformName.TvOS, 12,0)]
-		[Deprecated (PlatformName.WatchOS, 5,0)]
+#if !NET
+		[iOS (10,3), TV (10,3)]
+#else
+		[SupportedOSPlatform ("ios10.3")]
+		[SupportedOSPlatform ("tvos10.3")]
+#endif
+#if !NET
+		[Deprecated (PlatformName.iOS, 12, 0, message: "This method is no longer available.")]
+		[Deprecated (PlatformName.TvOS, 12, 0, message: "This method is no longer available.")] 
+		[Deprecated (PlatformName.WatchOS, 5, 0, message:  "This method is no longer available.")]
+#else
+		[UnsupportedOSPlatform ("ios12.0")]
+		[UnsupportedOSPlatform ("tvos12.0")]
+#if IOS
+		[Obsolete ("Starting with ios12.0 This method is no longer available.", DiagnosticId = "BI1234", UrlFormat = "https://github.com/xamarin/xamarin-macios/wiki/Obsolete")]
+#elif TVOS
+		[Obsolete ("Starting with tvos12.0 This method is no longer available.", DiagnosticId = "BI1234", UrlFormat = "https://github.com/xamarin/xamarin-macios/wiki/Obsolete")]
+#endif
+#endif
 		[DllImport (Constants.SecurityLibrary)]
 		static extern /* __nullable SecKeyRef */ IntPtr SecCertificateCopyPublicKey (IntPtr /* SecCertificateRef */ certificate);
 
-		[iOS (10,3)]
-		[TV (10,3)]
-		[Deprecated (PlatformName.iOS, 12,0, message: "Use 'GetKey' instead.")]
-		[Deprecated (PlatformName.TvOS, 12,0, message: "Use 'GetKey' instead.")]
-		[Deprecated (PlatformName.WatchOS, 5,0, message: "Use 'GetKey' instead.")]
+#if !NET
+		[iOS (10,3), TV (10,3)]
+#else
+		[SupportedOSPlatform ("ios10.3")]
+		[SupportedOSPlatform ("tvos10.3")]
+#endif
+#if !NET
+		[Deprecated (PlatformName.iOS, 12, 0, message: "This method is no longer available.")]
+		[Deprecated (PlatformName.TvOS, 12, 0, message: "This method is no longer available.")] 
+		[Deprecated (PlatformName.WatchOS, 5, 0, message:  "This method is no longer available.")]
```

Comment:
wrong message!

That's why keeping the original is important. It's much easier to spot a mistake where a message would be incorrectly updated.

Reply:
Message should be the one from the PInvoke.